src/core/mac/sub_mac.cpp

```diff
@@ -267,6 +280,11 @@ void SubMac::ProcessTransmitSecurity(void)
     VerifyOrExit(mTransmitFrame.GetTimeIeOffset() == 0, OT_NOOP);
 #endif
 
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    // Transmit security will be processed after time IE content is updated.
+    VerifyOrExit(mTransmitFrame.mInfo.mTxInfo.mCslPresent == 0, OT_NOOP);
+#endif
+
```

Comment:
```suggestion
```
For supporting CSL, radio platform needs to have capability `OT_RADIO_CAPS_TRANSMIT_SEC`. `ShouldHandleTransmitSecurity()` at line 258 already handles this.